daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -72,13 +72,21 @@ trait ElementBaseGrammarMixin
     }
   }
 
-  protected lazy val isDelimitedPrefixedPattern = {
+  protected lazy val isPrefixed: Boolean = {
+    import LengthKind._
+    lengthKind match {
+      case Prefixed => true
+      case _ => false
+    }
+  }
```

Comment:
This can just be `lengthKind eq LengthKind.Prefixed`. Don't really need a match/case if we are just going to return true/false.